File: esp_provisioning_ios/ios/Classes/AccessPointSecurity.swift
import Foundation

enum AccessPointSecurity: Int, Codable {
  case open = 0
  case wep = 1
  case wpaPsk = 2
  case wpa2Psk = 3
  case wpaWpa2Psk = 4
  case wpa2Enterprise = 5
  case unknown = 6

  func encode(to encoder: Encoder) throws {
    let stringValue: String
    var container = encoder.singleValueContainer()

    switch self {
    case .open: stringValue = "OPEN"
    case .wep: stringValue = "WEP"
    case .wpaPsk: stringValue = "WPA_PSK"
    case .wpa2Psk: stringValue = "WPA2PSK"
    case .wpaWpa2Psk: stringValue = "WPA_WPA2_PSK"
    case .wpa2Enterprise: stringValue = "WPA2_ENTERPRISE"
    case .unknown: stringValue = "UNKNOWN"
    }

    try container.encode(stringValue)
  }
}

File: esp_provisioning_android/android/src/main/kotlin/com/spindance/esp_provisioning/model/AccessPointSecurity.kt
package com.spindance.esp_provisioning.model

enum class AccessPointSecurity(val value: Int) {
  OPEN(0),
  WEP(1),
  WPA_PSK(2),
  WPA2PSK(3),
  WPA_WPA2_PSK(4),
  WPA2_ENTERPRISE(5),
  UNKNOWN(6);

  companion object {
    fun fromValue(value: Int) =
      AccessPointSecurity.values().firstOrNull { it.value == value} ?: UNKNOWN
  }
}
